packages/runtime/container-runtime/src/test/garbageCollection.spec.ts

```diff
@@ -257,323 +231,412 @@ describe("Garbage Collection Tests", () => {
+            };
 
-            // Remove node 2's reference from node 1. This should make node 2 and node 3 unreferenced.
-            defaultGCData.gcNodes[nodes[1]] = [];
+            it("doesn't generate events for referenced nodes", async () => {
+                const garbageCollector = createGCOverride();
 
-            await garbageCollector.collectGarbage({ runGC: true });
+                // Run garbage collection on the default GC data where everything is referenced.
+                await garbageCollector.collectGarbage({});
 
-            // Validate that no inactive events are logged yet.
-            await updateAllNodesAndRunGC(garbageCollector);
-            validateNoInactiveEvents();
+                // Advance the clock just before the timeout and validate no events are generated.
+                clock.tick(timeout - 1);
+                await updateAllNodesAndRunGC(garbageCollector);
+                validateNoUnexpectedEvents();
 
-            // Expire the unreferenced timer (if any).
-            clock.tick(inactiveTimeoutMs + 1);
+                // Advance the clock to expire the timeout.
+                clock.tick(1);
 
-            // Validate that changed and loaded events for node 2 and node 3 are logged since they are inactive.
-            await updateAllNodesAndRunGC(garbageCollector);
-            assert(
-                mockLogger.matchEvents([
-                    { eventName: changedEvent, timeout: inactiveTimeoutMs, id: nodes[2], pkg: eventPkg },
-                    { eventName: loadedEvent, timeout: inactiveTimeoutMs, id: nodes[2], pkg: eventPkg },
-                    { eventName: changedEvent, timeout: inactiveTimeoutMs, id: nodes[3], pkg: eventPkg },
-                    { eventName: loadedEvent, timeout: inactiveTimeoutMs, id: nodes[3], pkg: eventPkg },
-                ]),
-                "inactive events not logged as expected",
-            );
+                // Update all nodes again. Validate that no unexpected events are generated since everything is referenced.
+                await updateAllNodesAndRunGC(garbageCollector);
+                validateNoUnexpectedEvents();
+            });
 
-            // Add reference from node 1 to node 3 and validate that revived event is logged.
-            garbageCollector.addedOutboundReference(nodes[1], nodes[3]);
-            await garbageCollector.collectGarbage({});
-            assert(
-                mockLogger.matchEvents([
-                    { eventName: revivedEvent, timeout: inactiveTimeoutMs, id: nodes[3], pkg: eventPkg, fromId: nodes[1] },
-                ]),
-                "revived event not logged as expected",
-            );
-        });
+            it("generates events when nodes that are used after time out", async () => {
+                const garbageCollector = createGCOverride();
 
-        it("generates only revived event when an inactive node is changed and revived", async () => {
```

Comment:
Looks like this test is actually deleted - why?

Reply:
Good catch. This got deleted during merge with the previous PR (I had 2 in progress that modified the tests). Added it back.